README.md

```diff
@@ -132,108 +132,64 @@ Returns `true` if there exists an edge from node _u_ to node _v_. Returns `false
-
-<a name="outdegree" href="#outdegree">#</a> <i>graph</i>.<b>outdegree</b>(<i>node</i>)
-
-Computes the [outdegree](https://en.wikipedia.org/wiki/Directed_graph#Indegree_and_outdegree) (number of outgoing edges) for the specified _node_.
+Gets the adjacent node list for the specified node. The argument _node_ is a node reference (object or string). Returns a `Set` of adjacent node references or `undefined` if the node is not found.
 
 ### Serialization
 
-<a name="serialize" href="#serialize">#</a> <i>graph</i>.<b>serialize</b>()
+<a name="serializeGraph" href="#serializeGraph">#</a> <b>serializeGraph</b>(<i>graph</i>)
 
 Serializes the graph. Returns an object with the following properties.
 
-- `nodes` An array of objects, each with an `id` property whose value is a node identifier string.
+- `nodes` An array of objects, each representing a node reference.
 - `links` An array of objects representing edges, each with the following properties.
-  - `source` The node identifier string of the source node (**u**).
-  - `target` The node identifier string of the target node (**v**).
+  - `source` The node reference of the source node (**u**).
+  - `target` The node reference of the target node (**v**).
   - `weight` The weight of the edge between the source and target nodes.
 
 Here's example code for serializing a graph.
 
 ```javascript
-var graph = Graph();
+var graph = new Graph();
 graph.addEdge('a', 'b');
 graph.addEdge('b', 'c');
-var serialized = graph.serialize();
+var serialized = serializeGraph(graph);
 ```
 
-The following will be the value of `serialized`.
+<a name="deserializeGraph" href="#deserializeGraph">#</a> <b>deserializeGraph</b>(<i>serialized</i>)
 
-```json
-{
-  "nodes": [{ "id": "a" }, { "id": "b" }, { "id": "c" }],
-  "links": [
-    { "source": "a", "target": "b", "weight": 1 },
-    { "source": "b", "target": "c", "weight": 1 }
-  ]
-}
-```
-
-This representation conforms to the convention of graph representation when working with D3.js force layouts. See also [d3.simulation.nodes](https://github.com/d3/d3-force#simulation_nodes) and [d3.forceLinks](https://github.com/d3/d3-force#links).
-
-<a name="deserialize" href="#deserialize">#</a> <i>graph</i>.<b>deserialize</b>(<i>serialized</i>)
-
-Deserializes the given serialized graph. Returns _graph_ to support method chaining. The argument _serialized_ is a graph representation with the structure described in **[serialize](#serialize)**. This function iterates over the serialized graph and adds the nodes and links it represents by invoking **[addNode](#add-node)** and **[addEdge](#add-edge)**. The output from **[serialize](#serialize)** can be used as the input to **deserialize**.
+Deserializes the given serialized graph. Returns a new _graph_. The argument _serialized_ is a graph representation with the structure described in **[serializeGraph](#serializeGraph)**.
```

Comment:
Makes me think that the `serializeGraph` could also use an identity function, to avoid the duplication of the data when using a node object.